src/main/java/lv/ctco/javaschool/goal/boundary/GoalApi.java

```diff
@@ -126,6 +128,53 @@ public void saveNewCommentsForGoalById(@PathParam("id") Long goalId, MessageDto
                 .collect(Collectors.toList());
     }
 
+    @POST
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/search-goals")
+    public List<GoalDto> getGoalsByTag(JsonObject searchDto) {
+        List<GoalDto> goalDtoList = new ArrayList<>();
+        for (Map.Entry<String, JsonValue> pair : searchDto.entrySet()) {
+            String adr = pair.getKey();
+            String value = ((JsonString) pair.getValue()).getString();
+            if (adr.equals("goalsearch")) {
+                List<Tag> tagList = goalStore.getTagsByMessage(value);
+                for (Tag tag : tagList) {
+                    List<Goal> goalList = goalStore.getGoalsByTag(tag);
+                    if (goalList.size() != 0) {
```

Comment:
No need to check size before `stream()`.

Reply:
removed checking list size